src/librustc/ty/context.rs

```diff
@@ -685,6 +697,7 @@ impl<'a, 'gcx, 'tcx> TyCtxt<'a, 'gcx, 'tcx> {
     pub fn create_and_enter<F, R>(s: &'tcx Session,
                                   local_providers: ty::maps::Providers<'tcx>,
                                   extern_providers: ty::maps::Providers<'tcx>,
+                                  mir_passes: Rc<Passes>,
```

Comment:
Why are they an input and not hardcoded in `librustc_mir`? My intuition is that they will have to be interacting with some sort of scheduler within a function, interleaving "passes" with inlining.